core/node/consensus/src/storage/store.rs

```diff
@@ -444,44 +444,18 @@ impl storage::PersistentBatchStore for Store {
         self.batches_persisted.clone()
     }
 
-    /// Get the earliest L1 batch number which has to be signed by attesters.
-    async fn earliest_batch_number_to_sign(
+    /// Get the next L1 batch number which has to be signed by attesters.
+    async fn next_batch_to_attest(
         &self,
         ctx: &ctx::Ctx,
     ) -> ctx::Result<Option<attester::BatchNumber>> {
-        // This is the rough roadmap of how this logic will evolve:
-        // 1. Make best effort at gossiping and collecting votes; the `BatchVotes` in consensus only considers the last vote per attesters.
-        //    Still, we can re-sign more than the last batch, anticipating step 2.
-        // 2. Ask the Main Node what is the earliest batch number that it still expects votes for (ie. what is the last submission + 1).
-        // 3. Change `BatchVotes` to handle multiple pending batch numbers, anticipating that batch intervals might decrease dramatically.
-        // 4. Once QC is required to submit to L1, Look at L1 to figure out what is the last submission, and sign after that.
-
-        // Originally this method returned all unsigned batch numbers by doing a DAL query, but we decided it should be okay and cheap
-        // to resend signatures for already signed batches, and we don't have to worry about skipping them. Because of that, we also
-        // didn't think it makes sense to query the database for the earliest unsigned batch *after* the submission, because we might
-        // as well just re-sign everything. Until we have a way to argue about the "last submission" we just re-sign the last 10 to
-        // try to produce as many QCs as the voting register allows, within reason.
-
-        // The latest decision is not to store batches with gaps between in the database *of the main node*.
-        // Once we have an API to serve to external nodes the earliest number the main node wants them to sign,
-        // we can get rid of this method: on the main node we can sign from what `last_batch_qc` returns, and
-        // while external nodes we can go from whatever the API returned.
-
-        const NUM_BATCHES_TO_SIGN: u64 = 10;
-
-        let Some(last_batch_number) = self
+        Ok(self
             .conn(ctx)
             .await?
-            .get_last_batch_number(ctx)
+            .attestation_status(ctx)
             .await
-            .wrap("get_last_batch_number")?
-        else {
-            return Ok(None);
-        };
-
-        Ok(Some(attester::BatchNumber(
-            last_batch_number.0.saturating_sub(NUM_BATCHES_TO_SIGN),
-        )))
+            .wrap("next_batch_to_attest")?
+            .map(|s| s.next_batch_to_attest))
```

Comment:
once you rebase, remember to verify that genesis hash is consistent.

Reply:
This is retrieving the `AttestationStatus` from the database. Do you mean to also retrieve the genesis from the database and compare? How could it be different?

Reply:
scenario: the main node restarts, reverts some batches, does regenesis, we don't notice that the next_batch_to_attest moved back, the era-consensus code throws some error, the whole node shuts down with a different error than it should (should be "genesis changed", while we get "next_batch_to_attest decreased" or some even more enigmatic error, depending on what exactly fails).

It is not a huge difference, just preventing some garbage getting into the system.

Reply:
Okay there are certainly some misunderstandings on my part here: I thought that to do a revert the main node is stopped, some revert CLI is executed, a new genesis is configured, the main node is started, it does [regenesis, and then it starts](https://github.com/matter-labs/zksync-era/blob/c4d8c5266610deda3d43d626191be8c766c9835f/core/node/consensus/src/mn.rs#L39) the stores and the attestation status. If the genesis could change then the `BlockStore` would be in trouble as it caches it in the beginning. 

Another problem is that the `AttesterStatusRunner` simply retries after an error, the only thing it doesn't do is not update the status. Maybe it should have an option to trait errors as fatal, or let the client return a different result to indicate that the whole thing is borked, or do the check itself. This looked fine for the EN as it will eventually stop when it sees the genesis changed.

I'll open another PR that changes the `AttestationStatusClient` to return the whole status so this can be inspected and differentiated from RPC errors.

Reply:
> the main node is started, it does regenesis, and then it starts

that's true. Then EN observe the genesis of the main node and restart if it changes. But it is not atomic.

> Maybe it should have an option to trait errors as fatal

I see, I think that's ok for it not to be fatal. Then this is issue is even less important, as the wrong error will be only logged.

Reply:
https://github.com/matter-labs/era-consensus/pull/167